esp/services/esdl_svc_engine/esdl_svc_engine.cpp

```diff
@@ -103,6 +103,22 @@ IPropertyTree *createContextMethodConfig(IPropertyTree *methodConfig)
     return LINK(methodConfig); //no copy, nothing changed
 }
 
+bool skipContextConfig(IPropertyTree *cfg)
+{
+    if (!cfg->getPropBool("@contextConfig", true)) //explicitly disabled
+        return true;
+
+    const char *elInclude = cfg->queryProp("@contextInclude");
+    const char *elRemove = cfg->queryProp("@contextRemove");
+    const char *attRemove = cfg->queryProp("@contextAttRemove");
+
+    bool noElIncluded = elInclude && !*elInclude; //empty xpath
+    bool allElRemoved = (elRemove && streq(elRemove, "*"));
+    bool allAttRemoved = (attRemove && streq(attRemove, "*"));
+
+    return (allAttRemoved && (noElIncluded || allElRemoved));
```

Comment:
I can't see a logical use case for an explicit request to omit all elements to be completely ignored because I did not also include a separate explicit request to omit all parts of all elements. That's essentially what this is saying: contextRemove can omit everything but only if contextAttRemove explicitly confirms the intent to omit everything, and absent this confirmation nothing is omitted.

Maybe if attRemove is not empty and not "*", but even that seems like a stretch to override elRemove.

In what scenario would someone specify contextRemove="*" and not expect everything to be removed?

Reply:
@timothyklemm this isn't about what are the likely scenarios, it's about making sure that there is a proper syntax for anything the user wants to do AND is backward compatible by default.

contextAttRemove doesn't affect attributes at every level, it only affects top level attributes, and contextRemove only affects child elements.

I am treating them as ptree xpaths with additional support for "|".

It's done this way to make sure I can support "remove all child elements" but am able keep top level attributes.  
Defaulting contextAttRemove to "*" would not be backward compatible, and it would be hard to then say "keep all".

When users want to build highly custom contexts they can use a request transform.

Reply:
If contextAttRemove ="*" were the default, keep all could be contextAttRemove="".  But that still doesn't help with backward compatibility.

Reply:
I suppose if we don't care about logical consistency I could say that contextRemove="*" changes the default for contextAttRemove to "*" also.  Then to keep all the user would have to set contextAttRemove="".  It's weird but would be convenient.

Reply:
As we discussed offline I will use contextRemove="." to mean remove the entire current (root) node. Which will include the attributes.  I will remove contextConfig="true/false" since that was meant as a similar simplification.